### PR TITLE
Leverage shared package for zip deploy

### PR DESCRIPTION
**Pull request by EricJizbaMSFT**
Here's the package: https://github.com/EricJizbaMSFT/vscode-azure-sdk

Obviously we'll have to discuss if that shared package will work and find an actual home for it instead of my fork. For now, I only moved the zip logic and start/stop. Eventually we should move more logic over.

### CODE REVIEW

File: src/explorer/siteNodeBase.ts [hunk 1]
```diff
@@ -56,29 +59,11 @@ export class SiteNodeBase extends NodeBase {
...
-        const siteName = util.extractSiteName(this.site);
-
-        if (util.isSiteDeploymentSlot(this.site)) {
-            const slotName = util.extractDeploymentSlotName(this.site);
-            await this.webSiteClient.webApps.startSlot(rgName, siteName, slotName);
-        } else {
-            await this.webSiteClient.webApps.start(rgName, siteName);
-        }
-        await util.waitForWebSiteState(this.webSiteClient, this.site, 'running');
+        this._siteWrapper.start(this.webSiteClient);
```
Comment (StephenWeatherford) [inline src/explorer/siteNodeBase.ts]: Shouldn't this do an await?

File: src/explorer/siteNodeBase.ts [hunk 2]
```diff
@@ -56,29 +59,11 @@ export class SiteNodeBase extends NodeBase {
...
-        const siteName = util.extractSiteName(this.site);
-
-        if (util.isSiteDeploymentSlot(this.site)) {
-            const slotName = util.extractDeploymentSlotName(this.site);
-            await this.webSiteClient.webApps.stopSlot(rgName, siteName, slotName);
-        } else {
-            await this.webSiteClient.webApps.stop(rgName, siteName);
-        }
-        await util.waitForWebSiteState(this.webSiteClient, this.site, 'stopped');
+        this._siteWrapper.stop(this.webSiteClient);
```
Comment (StephenWeatherford) [inline src/explorer/siteNodeBase.ts]: ditto

File: package.json [hunk 3]
```diff
@@ -371,7 +371,8 @@
         "opn": "^5.1.0",
         "request": "^2.83.0",
         "simple-git": "^1.77.0",
-        "vscode-extension-telemetry": "^0.0.6"
+        "vscode-extension-telemetry": "^0.0.6",
+        "vscode-azure-appservice": "https://github.com/EricJizbaMSFT/vscode-azure-sdk/releases/download/appservice-v0.1.1/vscode-azure-appservice-0.1.1.tgz"
```
Comment (whiskeyjay) [inline package.json]: Will you switch to the "official" repo with this check-in?

Comment (EricJizbaMSFT) [reply package.json]: Yes - I won't merge these PRs until the official repo is created, reviewed, and I've published the packages to npmjs.

**Comment by EricJizbaMSFT** [comment]
Okay the PR for the shared packages is out: https://github.com/Microsoft/vscode-azuretools/pull/1

This PR is updated with latest from master and I removed link to package in my custom fork. CI build will fail until shared package is published.

**Pull request merged**